src/append/rolling_file/mod.rs

```diff
@@ -429,49 +465,146 @@ appenders:
+        fn process(&self, _: &mut LogFile) -> anyhow::Result<()> {
+            Ok(())
+        }
+        fn is_pre_process(&self) -> bool {
+            true
+        }
+    }
+
     #[test]
-    fn append() {
-        let dir = tempfile::tempdir().unwrap();
-        let path = dir.path().join("append.log");
-        RollingFileAppender::builder()
-            .append(true)
-            .build(&path, Box::new(NopPolicy))
-            .unwrap();
-        assert!(path.exists());
-        File::create(&path).unwrap().write_all(b"hello").unwrap();
-
-        RollingFileAppender::builder()
-            .append(true)
-            .build(&path, Box::new(NopPolicy))
-            .unwrap();
-        let mut contents = vec![];
-        File::open(&path)
-            .unwrap()
-            .read_to_end(&mut contents)
-            .unwrap();
-        assert_eq!(contents, b"hello");
+    fn test_rolling_append() {
+        use log::Level;
+
+        let tmp_file = NamedTempFile::new().unwrap();
+        let policies: Vec<Box<dyn Policy>> = vec![Box::new(NopPrePolicy), Box::new(NopPostPolicy)];
+        let record = Record::builder()
+            .level(Level::Debug)
+            .target("target")
+            .module_path(Some("module_path"))
+            .file(Some("file"))
+            .line(Some(100))
+            .build();
+        log_mdc::insert("foo", "bar");
+
+        for policy in policies {
+            let appender = RollingFileAppender::builder()
+                .append(true)
+                .encoder(Box::new(PatternEncoder::new("{m}{n}")))
+                .build(&tmp_file.path(), policy)
+                .unwrap();
+
+            assert!(appender.append(&record).is_ok());
+
+            // No-op method, but get the test coverage :)
+            appender.flush();
+        }
     }
 
     #[test]
-    fn truncate() {
-        let dir = tempfile::tempdir().unwrap();
-        let path = dir.path().join("truncate.log");
-        RollingFileAppender::builder()
-            .append(false)
-            .build(&path, Box::new(NopPolicy))
-            .unwrap();
-        assert!(path.exists());
-        File::create(&path).unwrap().write_all(b"hello").unwrap();
-
-        RollingFileAppender::builder()
-            .append(false)
-            .build(&path, Box::new(NopPolicy))
-            .unwrap();
-        let mut contents = vec![];
-        File::open(&path)
-            .unwrap()
-            .read_to_end(&mut contents)
-            .unwrap();
-        assert_eq!(contents, b"");
+    fn test_logfile() {
+        let tmp_file = NamedTempFile::new().unwrap();
+        let mut logfile = LogFile {
+            writer: &mut None,
+            path: tmp_file.path(),
+            len: 0,
+        };
+
+        assert_eq!(logfile.path(), tmp_file.path());
+        assert_eq!(logfile.len_estimate(), 0);
+
+        // No actions to take here, the writer becomes inaccessible but theres
+        // no getter to verify
+        logfile.roll();
```

Comment:
cant we verify the roll has happened?

Reply:
Hmm yeah, I can create a dummy LogWriter and make sure it gets cleaned up. It's a tough line of UTs without proper Mocks but that's a future addition.